src/net_processing.cpp

```diff
@@ -640,6 +652,9 @@ class PeerManagerImpl final : public PeerManager
     void UpdateBlockAvailability(NodeId nodeid, const uint256& hash) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
     bool CanDirectFetch() EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 
+    void UpdateKnownChainTips(NodeId node_id, const CBlockIndex& new_index);
+    bool IsIndexOnKnownChain(NodeId node_id, const CBlockIndex& index);
```

Comment:
This function could be introduced in commit 0fdd0eb66a85189f2c32c33defb4322fd2a914dd (*[net processing] Prevent fingerprinting when receiving header*), rather than the previous commit.